src/rust/engine/process_execution/src/local.rs

```diff
@@ -709,89 +735,44 @@ pub async fn prepare_workdir(
       if let Some(working_directory) = &req.working_directory {
         executable_path = working_directory.as_ref().join(executable_path)
       }
-      Some(executable_path)
+      Some(workdir_path.join(executable_path))
```

Comment:
Was this a bug-fix picked up along the way? Afaict this is new over and above the shuffle on the LHS.

Reply:
Not a bug fix: we have a good test for this. I just pushed up the joining of the workdir_path to avoid an unnecessary clone.